python/lsst/daf/butler/core/location.py

```diff
@@ -368,6 +368,24 @@ def pathInStore(self):
         """
         return self._path
 
+    @property
+    def bucketName(self):
+        """If Location is an S3 protocol, returns the bucket name."""
+        if self._datastoreRootUri.scheme == 's3':
+            return self._datastoreRootUri.netloc
+        else:
+            raise AttributeError(f'Path {self} is not a valid S3 protocol.')
+
+    @property
+    def pathInBucket(self):
+        """Returns the path in an S3 Bucket, normalized so that directory
+        structure in the bucket matches that of a local filesystem.
```

Comment:
But it doesn't match that of a local filesystem because the root directory is stripped.  This is actually the path of the location that includes the datastore and the path in datastore but is relative to the root directory. That is a generic concept but I still don't understand why it's needed.  Renaming this to `pathRelativeToTopLevel` would be closer (But "root" and "toplevel" are overloaded in `Location` because they already confuse with "root of the datastore" as opposed to "root of the file system").